### PR TITLE
Add ability not to compile fabio from sources

### PR DESCRIPTION
**Pull request by kostyrev**
- optionally run under dedicated user
- optionally install from compiled binary
- provide upstart script for CentOS

### CODE REVIEW

File: tasks/install.yml [hunk 1]
```diff
@@ -18,45 +18,18 @@
...
+- include: install_binary.yml
+  when: not fabio_install_from_source
 
 - name: Install Fabio service
   become: yes
   template:
     dest: /etc/init/fabio.conf
     mode: 0755
-    src: fabio.conf.j2
+    src: "{{ 'fabio.centos.upstart.j2' if ansible_distribution == 'CentOS' and ansible_distribution_version|version_compare(7, '<') else 'fabio.conf.j2' }}"
```
Comment (lobsterdore) [inline tasks/install.yml]: Can you split this into two separate tasks that are triggered by when clauses?

File: tasks/install_binary.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+
+- name: Get fabio checksums file
```
Comment (lobsterdore) [inline tasks/install_binary.yml:3]: I think you can replace this and the shell below with the uri module with return_content set to yes, you will need to ensure that httplib2 is installed first though.

File: tasks/install.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 - name: Install required packages
   become: yes
-  apt:
+  package:
```
Comment (lobsterdore) [inline tasks/install.yml:5]: This breaks versions before 2.x.

File: tasks/main.yml [hunk 4]
```diff
@@ -1,5 +1,9 @@
 ---
 
+- name: Create fabio user
+  include: user.yml
```
Comment (lobsterdore) [inline tasks/main.yml]: Can you move this include into install.yml? I would like to keep install and configure as the main task files.

File: defaults/main.yml [hunk 5]
```diff
@@ -1,5 +1,18 @@
 ---
 
+fabio_create_user: false
```
Comment (lobsterdore) [inline defaults/main.yml]: Can you order all the vars in this file alphabetically?

File: tasks/install.yml (hunk 3 above)
Comment (kostyrev) [reply tasks/install.yml:5]: yeah, I know. But is there any reason why should someone still use Ansible < 2 that was released more than 6 months ago.

File: tasks/install.yml (hunk 1 above)
Comment (kostyrev) [reply tasks/install.yml]: but then there will be 5 duplicated lines.
I'll think of something.

File: tasks/main.yml (hunk 4 above)
Comment (kostyrev) [reply tasks/main.yml]: done

File: defaults/main.yml (hunk 5 above)
Comment (kostyrev) [reply defaults/main.yml]: done

File: tasks/install.yml (hunk 1 above)
Comment (kostyrev) [reply tasks/install.yml]: done

File: tasks/install_binary.yml (hunk 2 above)
Comment (kostyrev) [reply tasks/install_binary.yml:3]: I didn't get it. Why?
Yes, uri module with return_content will get me a page with checksums of binaries for all platforms.
Still somehow I'll need to extract checksum for only one for fabio-1.3.4-go1.7.3-linux_amd64

FYI
>The dependency on httplib2 was removed in Ansible 2.1

File: tasks/install.yml (hunk 3 above)
Comment (lobsterdore) [reply tasks/install.yml:5]: Very true, in that case can you remove the 1.9.4 refs as part of this PR? There is a reference in travis.yml and the min version in meta/main.yml.

File: tasks/install_binary.yml (hunk 2 above)
Comment (lobsterdore) [reply tasks/install_binary.yml:3]: Yep you're correct, ignore my comment here.

File: tasks/install.yml (hunk 3 above)
Comment (kostyrev) [reply tasks/install.yml:5]: done

**Comment by lobsterdore** [comment]
LGTM :+1:

**Pull request merged**